pyuvdata/uvbase.py

```diff
@@ -356,8 +358,34 @@ def check(
                                         " be: " + str(param.expected_type)
                                     )
                         else:
-                            # Array
-                            if not isinstance(param.value.item(0), param.expected_type):
+                            # Array or quantity
+                            if isinstance(param.value, Quantity):
+                                # check if user put expected type as a type of quantity
+                                # not a more generic type of number.
+                                if any(
+                                    issubclass(param_type, Quantity)
+                                    for param_type in _get_iterable(param.expected_type)
+                                ):
+                                    # Verify the param is an instance
+                                    # of the specific Quantity type
+                                    if not isinstance(param.value, param.expected_type):
+                                        raise ValueError(
+                                            f"UVParameter {p} is a Quantity object "
+                                            "but not the appropriate type. "
+                                            f"Is {type(param.value)} but "
+                                            f"expected {param.expected_type}."
+                                        )
+                                    else:
+                                        # matches expected type
+                                        continue
+                                else:
+                                    # Expected type is not a Quantity subclass
+                                    # Assuming it is a data type like float, int, etc
+                                    # continuing with check below
+                                    check_val = param.value.item(0).value
```

Comment:
Ok, now I think I understand what you're doing but I'm not sure if it's what we want. I think this will allow a situation to pass where you specified that the expected type was a float and the actual parameter is a Quantity. Do we want that to pass?

Reply:
Right that's with what would happen. I had the though that choice was up to the user :thinking:

Quantities definitely add a layer of complexity. This would let you like "upgrade" a parameter to a quantity (like the situation you detailed), but never the other way if the type is expected to be a quantity specifically. I personally would prefer to have maybe a new type of parameter specifically for quantites that would track the expected type like it is now, but maybe also the equivalency class of the unit. But not sure that was really in the scope here, and probably needs more flushing out anyway.